src/CoreWCF.ConfigurationManager/tests/CustomBindingTests.cs

```diff
@@ -0,0 +1,780 @@
+            {
+                using (ServiceProvider provider = CreateProvider(fs.Name))
+                {
+                    IConfigurationHolder settingHolder = GetConfigurationHolder(provider);
+
+                    CustomBinding actualBinding = settingHolder.ResolveBinding(nameof(CustomBinding), expectedName) as CustomBinding;
+
+                    Assert.Equal(expectedName, actualBinding.Name);
+                    Assert.Equal(expectedDefaultTimeout, actualBinding.CloseTimeout);
+                    Assert.Equal(expectedDefaultTimeout, actualBinding.OpenTimeout);
+                    Assert.Equal(expectedDefaultTimeout, actualBinding.SendTimeout);
+                    Assert.Equal(expectedReceiveTimeout, actualBinding.ReceiveTimeout);
+                }
+            }
+        }
+
+        [Fact]
+        public void CustomBinding_WithDefaultSettings()
+        {
+            string expectedName = "customBinding";
+            TimeSpan expectedReceiveTimeout = TimeSpan.FromMinutes(10);
+            TimeSpan expectedDefaultTimeout = TimeSpan.FromMinutes(1);
+
+
+            string xml = $@"
+<configuration> 
+    <system.serviceModel>         
+        <bindings>         
+            <customBinding>
+                <binding name=""{expectedName}"">
+                </binding>
+            </customBinding>                             
+        </bindings>                             
+   </system.serviceModel>
+</configuration>";
+
+            using (var fs = TemporaryFileStream.Create(xml))
+            {
+                using (ServiceProvider provider = CreateProvider(fs.Name))
+                {
+                    IConfigurationHolder settingHolder = GetConfigurationHolder(provider);
+
+                    CustomBinding actualBinding = settingHolder.ResolveBinding(nameof(CustomBinding), expectedName) as CustomBinding;
+
+                    Assert.Equal(expectedName, actualBinding.Name);
+                    Assert.Equal(expectedDefaultTimeout, actualBinding.CloseTimeout);
+                    Assert.Equal(expectedDefaultTimeout, actualBinding.OpenTimeout);
+                    Assert.Equal(expectedDefaultTimeout, actualBinding.SendTimeout);
+                    Assert.Equal(expectedReceiveTimeout, actualBinding.ReceiveTimeout);
+                }
+            }
+        }
+
+        #endregion
+
+        #region Encoding
+
+        #region TextEncoding
+
+        private const string UnicodeBe = "UnicodeBE";
+        private const string Unicode = "unicode";
+        private const string Utf8 = "utf-8";
+        private static Encoding GetEncodingForTest(string encodingName)
+            => encodingName switch
+            {
+                UnicodeBe => Encoding.BigEndianUnicode,
+                Utf8 => Encoding.UTF8,
+                Unicode => Encoding.Unicode,
+                _ => Encoding.Default
+            };
+
+        [Theory]
+        [InlineData(UnicodeBe)]
```

Comment:
Have a look at the MemberData attribute. You can pass it the name of a static method which returns an IEnumerable of test parameters. It allows you to pass instances instead of constants. Take a look at [this usage](https://github.com/CoreWCF/CoreWCF/blob/main/src/CoreWCF.Http/tests/BasicHttpTransportMessage.cs#L42) for an example. That way you can take an actual Encoding class as input to your test.